File: interview-Preps/microsoft/AddAllDigits.java
class AddAllDigits {
	public static void main(String args[]) {
        System.out.print(add(120));
	}

    static int add(int data){
        String digits  = String.valueOf(data);
        int length = digits.length(); 
        if (length == 1) return Integer.parseInt( "" + digits.charAt(0));
        else{
            int sum = 0;
            for(int i =0 ; i < digits.length() ; i++){
                sum += Integer.parseInt( "" + digits.charAt(i));
            }
            return add(sum);
        }
    }
}

File: challenges/microsoft/AddAllDigits.js
function addAllDigits(num){
  let stringArr = num.toString().split("");
  let length = stringArr.length;
  if(length == 1) return stringArr[0];
  else{
    let sum = 0;
    for(let i=0; i < length; i++){
      sum += parseInt(stringArr[i]);
    }
    return addAllDigits(sum);
  }
}
console.log(addAllDigits(2038));
